### ISSUE TITLE
Reduced-precision (bf16/fp16) decode produces garbage in C++ runtime for vision_language/embed_input bundles

### ISSUE THREAD

**Issue by samli-gpu** [NVIDIA/TensorRT-Model-Connect#182]
## Summary

The native C++ runtime (`trtmc run`) produces **garbage output at `--precision bf16` and `fp16`** for **vision-language / `embed_input` decoder bundles** (the non-split, single decoder-engine path). The **TRT engine itself is correct** at those precisions — the same bundle runs coherently through the pure-Python runner (`tensorrt_model_connect.debug_runner`) — so this is a **runtime execution bug**, not a builder/engine numerics problem. `--precision fp32` works correctly end-to-end.

This is **not model-specific**. It reproduces on stock `Qwen/Qwen2.5-VL-3B-Instruct`. (Found while adding a new `vision_language` family; that family is pinned to fp32 as a workaround.)

## Reproduce

```bash
# bf16: garbage
./build/trtmc build Qwen/Qwen2.5-VL-3B-Instruct -o /tmp/q.trtfb --max-cache-length 384 --precision bf16
./build/trtmc run /tmp/q.trtfb --prompt "What color is the vehicle? Answer in one word." \
  --image tests/e2e/data/test_img.jpeg --max-new-tokens 12 --greedy
# -> "This weekend problem solving for future information below the problem>SOC"

# fp32: correct
./build/trtmc build Qwen/Qwen2.5-VL-3B-Instruct -o /tmp/q32.trtfb --max-cache-length 384 --precision fp32
./build/trtmc run /tmp/q32.trtfb --prompt "What color is the vehicle? Answer in one word." \
  --image tests/e2e/data/test_img.jpeg --max-new-tokens 12 --greedy
# -> correct, concise answer
```

The same **bf16** `engine_plan` from the bundle runs **coherently** via Python:

```python
from tensorrt_model_connect.debug_runner import runner_from_bundle
r = runner_from_bundle("/tmp/q.trtfb")
r.generate(input_ids, 12)   # coherent text
```

## What is ruled out

Verified during investigation that the following are **not** the cause:

- **Tokenizer** — the C++ native BPE emits the same ids as HuggingFace.
- **Logits dtype** — logits are cast to fp32 in the builder; the runtime reads fp32 correctly.
- **KV-cache write** — byte-verified that the per-step present→cache copy is correct (`present_bytesum == cache_bytesum` at every step; element size = 2 for bf16).
- **Attention mask / position ids** — correct (valid cache columns + current token unmasked; positions 0..N-1).
- **CUDA graphs** — failure persists with `--set runtime.disable_cuda_graph=true`.
- **Optimization-profile detection** — handled by the runtime.

The TRT engine, its KV cache contents, and all engine inputs are verified correct; the **same bf16 engine produces correct logits in Python** with the same inputs and the same (byte-identical) cache.

## Localization

Divergence begins at the **first KV-cache-using step** (position ≥ 1) and the wrong logits are **finite (no NaN/Inf)**. Position 0 (no cache) matches the reference exactly. This points to a reduced-precision execution-context subtlety in the C++ non-split decode path (single decoder engine driven token-by-token), not present in the split-engine path used by plain decoders (e.g. Qwen3 at bf16 works, because it uses the separate prefill + decode engines).

## Impact

Any `vision_language` / `embed_input` family (which builds a single non-split decoder engine) is effectively limited to **fp32** through the native runtime at runtime. Plain `decoder_kv_cache` models that use the split prefill/decode engines are unaffected.

## Environment

- TensorRT 10.16.1.11, CUDA 13, Ubuntu 24.04, x86_64, NVIDIA RTX 6000 Ada (repository dev container image).
- Reproduced with the source-tree `./build/trtmc` (Release).

## Suggested next step

A runtime-internals diff between the C++ execution context and the Python `debug_runner` for the same single-profile/non-split decoder engine at bf16 (e.g. dumping per-layer engine activations inside the C++ context) should pinpoint the diverging op.

**Comment by samli-gpu** [comment]
fixed by PR #184
